Visualisation/js/main.js

```diff
@@ -0,0 +1,83 @@
+    maxSelectedSongs: 10
+}
+
+// Define variables (all variables have a getter and setter function so that related functions can be called when the value is changed). 
+// IMPORTANT: Never change the value of a variable directly, always use the setter function.
+
+let selectedSongs = [];
+const setSelectedSongs = (newSelectedSongs) => {
+    // Reassign colors
+    newSelectedSongs.forEach((song, index) => {
+        song.color = config.songColors[index % config.songColors.length];
+    }
+    );
+
+    selectedSongs = newSelectedSongs;
+
+    // Execute rerenders
+    renderMusicSelection();
+};
+
+let currentHoveredSongId = undefined;
+const setCurrentHoveredSongId = (newSongId) => {
+
+    // Update the border of the song image
+    if (currentHoveredSongId !== undefined && currentHoveredSongId !== newSongId) {
+        document.getElementById(`musicSelection-${currentHoveredSongId}Image`).style.border = `8px solid ${selectedSongs.find(song => song.id === currentHoveredSongId).color}`;
+    }
+
+    if (newSongId !== undefined) {
+        document.getElementById(`musicSelection-${newSongId}Image`).style.border = '8px solid #000000';
+    }
+
+    currentHoveredSongId = newSongId
+}
+
+
+let currentSearchTerm = '';
+const setCurrentSearchTerm = (searchTerm) => {
+    currentSearchTerm = searchTerm;
+    renderSearchResults();
+    document.getElementById('searchTerm').value = searchTerm;
+};
+
+let songs = [];
+const setSongs = (songsData) => {
+    songs = songsData;
+};
+
+// Define some commonly used state-mutation functions (functions that use the setter functions to change the state)
+const removeSong = (songId) => {
+    // Remove the song from the selection
+    const newSelection = selectedSongs.filter(song => song.id !== songId);
+    setSelectedSongs(newSelection);
+};
+
+const addSong = (songId) => {
+    if (selectedSongs.some(song => song.id === songId)) {
+        return;
+    }
+
+    // Add the song to the selection
+    const newSong = songs.find(song => song.id === songId);
+    setSelectedSongs([...selectedSongs, newSong]);
+};
+
+// Define data loading, try not to load the data multiple times.
+fetch('data/songs.json')
+    .then(response => response.json())
+    .then(songs => {
+        setSelectedSongs(songs.sort(() => 0.5 - Math.random()).slice(0, 10));
+        setSongs(songs);
+
+    })
+    .catch(error => console.error('Error loading songs:', error));
+
+renderMusicSelection();
```

Comment:
renderMusicSelection() is called here and in menu.js at l:64, both are only called for initializing the list right? both seem redundant to me (its also called in setSelectedSongs() ), but correct me if im wrong 😃

Reply:
You're absolutely right! Well spotted!